src/Compilers/Core/Portable/Syntax/SyntaxListBuilder.cs

```diff
@@ -81,15 +83,15 @@ public void AddRange(SyntaxList<SyntaxNode> list)
 
         public void AddRange(SyntaxList<SyntaxNode> list, int offset, int count)
         {
-            if (_nodes == null || this.Count + count > _nodes.Length)
+            if (this.Count + count > _nodes.Length)
             {
                 this.Grow(Count + count);
             }
 
             var dst = this.Count;
             for (int i = offset, limit = offset + count; i < limit; i++)
             {
-                _nodes[dst].Value = list.ItemInternal(i).Green;
+                _nodes[dst].Value = list.ItemInternal(i)!.Green;
```

Comment:
💡 Consider splitting the expression `list.ItemInternal(i)` to a local, then asserting that it is not null.